core/services/job/spawner_test.go

```diff
@@ -294,8 +294,9 @@ func TestSpawner_CreateJobDeleteJob(t *testing.T) {
 		chain := evmtest.MustGetDefaultChain(t, legacyChains)
 
 		evmRelayer, err := evmrelayer.NewRelayer(lggr, chain, evmrelayer.RelayerOpts{
-			DS:             db,
-			CSAETHKeystore: keyStore,
+			DS:                   db,
+			CSAETHKeystore:       keyStore,
+			CapabilitiesRegistry: capabilities.NewRegistry(lggr),
```

Comment:
there does not seem to be any test for the new feature. what's going on?

Reply:
technically this is testing the new check " c.CapabilitiesRegistry == nil " ;)

Reply:
@krehermann I've added some validation tests, which covers the case. I won't add tests to the custom provider implementation because when I started to I ended up mocking everything under the moon so I didn't feel like the test would add any value. Lmk if that works